### ISSUE TITLE
Use a Set collection for the mod blacklist/whitelist

### ISSUE THREAD

**Issue by coloursofnoise** [EverestAPI/Everest#575]
Currently these use lists, but I don't really see any reason why we'd want any duplicates in them

src: https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Mod/Everest/Everest.Loader.cs#L27-L57

**Comment by mayromr** [comment]
Hey, I was looking into this issue, It seems like a good idea to move to `HashSet`, But I have tried playing around with this change but it seems that `HashSet` doesn't implement `AsReadOnly`.
After looking online for a solution I found out that in newer [versions of .NET](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1?view=netframework-4.5.2#remarks) HashSet Implements `IReadOnlyCollection<T>` which will solve our issue, are there any plans (or is it even possible) to upgrade the .NET versions of Everest?

**Comment by Kalobi** [comment]
@nhruo123 Yes, we are in fact working on porting Everest to .NET 7.0 (see the `core` branch in this repo). A public beta should roll out soon.

**Comment by microlith57** [comment]
now that some time has passed, @Popax21 thoughts?
